src/modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp

```diff
@@ -17,7 +17,8 @@ namespace sndio {
 SoxSink::SoxSink(core::IAllocator& allocator, const Config& config)
     : output_(NULL)
     , buffer_(allocator)
-    , buffer_size_(config.frame_size)
+    , buffer_size_(
+          packet::ns_to_size(config.frame_length, config.sample_rate, config.channels))
```

Comment:
Nitpick: here and in SoxSource, I suggest to set `buffer_size_` to zero in constructor to make it clear that it can't be used until `setup_buffer_()` is called.

Reply:
done.